clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -625,7 +627,9 @@ public void commitAsync(Map<TopicPartition, OffsetAndMetadata> offsets, OffsetCo
     }
 
     // Visible for testing
-    CompletableFuture<Void> commit(final Map<TopicPartition, OffsetAndMetadata> offsets, final boolean isWakeupable) {
+    CompletableFuture<Void> commit(final Map<TopicPartition, OffsetAndMetadata> offsets,
+                                   final boolean isWakeupable,
+                                   final Optional<Long> timeoutMs) {
```

Comment:
This param is a bit subtile because it means two things: 1) retry and 2) the max duration. It is not obvious when you read it.

Reply:
Agree, maybe `retryExpirationTimeoutMs` better?